### ISSUE TITLE
[5.8.1] segfault in binmode STDOUT, ':stdio'; print 1

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6846]
Migrated from [rt.perl.org#24225](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=24225) (status was 'resolved')

Searchable as RT24225$

**Comment by p5pRT** [comment]
### From stas@stason.org

#### Created by stas@stason.org



% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
Segmentation fault

The same problem appears with blead perl\, but not with 5\.8\.0\.

\#0  0x40123024 in PerlIOStdio\_mode \(mode=0x0\,
&nbsp;     tmode=0xbffff208 "ØÒ\\001@&#8203;0×\\b@&#8203;$òÿ¿\\001"\) at perlio\.c&#8203;:2616
2616        while \(\*mode\) \{
\(gdb\) where
\#0  0x40123024 in PerlIOStdio\_mode \(mode=0x0\,
&nbsp;     tmode=0xbffff208 "ØÒ\\001@&#8203;0×\\b@&#8203;$òÿ¿\\001"\) at perlio\.c&#8203;:2616
\#1  0x401230e9 in PerlIOStdio\_pushed \(f=0x8054900\, mode=0x0\,
arg=0x40157618\,
&nbsp;     tab=0x40155a60\) at perlio\.c&#8203;:2640
\#2  0x4012086d in PerlIO\_push \(f=0x8054900\, tab=0x40155a60\, mode=0x0\,
&nbsp;     arg=0x40157618\) at perlio\.c&#8203;:1165
\#3  0x40120abc in PerlIO\_apply\_layera \(f=0x8054900\, mode=0x0\,
&nbsp;     layers=0x8054ad8\, n=0\, max=1\) at perlio\.c&#8203;:1248
\#4  0x40120b3c in PerlIO\_apply\_layers \(f=0x8054900\, mode=0x0\,
&nbsp;     names=0x805d310 "&#8203;:stdio"\) at perlio\.c&#8203;:1266
\#5  0x40120bbf in PerlIO\_binmode \(f=0x8054900\, iotype=62\, mode=0\,
&nbsp;     names=0x805d310 "&#8203;:stdio"\) at perlio\.c&#8203;:1291
\#6  0x400eae56 in Perl\_pp\_binmode \(\) at pp\_sys\.c&#8203;:755
\#7  0x4008d1f4 in Perl\_runops\_debug \(\) at dump\.c&#8203;:1434
\#8  0x400383c4 in S\_run\_body \(oldscope=1\) at perl\.c&#8203;:1829
\#9  0x40037e81 in perl\_run \(my\_perl=0x804baf8\) at perl\.c&#8203;:1748
\#10 0x080492b2 in main \(argc=3\, argv=0xbffff504\, env=0xbffff514\)
&nbsp;     at perlmain\.c&#8203;:86
\#11 0x401ebc57 in \_\_libc\_start\_main \(\) from /lib/i686/libc\.so\.6

<details><summary>Perl Info</summary>

```

Flags:
     category=core
     severity=medium

Site configuration information for perl v5.8.1:

Configured by stas at Thu Sep 25 19:12:52 PDT 2003.

Summary of my perl5 (revision 5.0 version 8 subversion 1) configuration:
   Platform:
     osname=linux, osvers=2.4.21-0.18mdkcustom, archname=i686-linux
     uname='linux rabbit.stason.org 2.4.21-0.18mdkcustom #6 mon jun 16 
16:26:34 est 2003 i686 unknown unknown gnulinux '
     config_args='-des -Dprefix=/home/stas/perl/5.8.1 -Doptimize=-g 
-Duseshrplib -Dusedevel'
     hint=recommended, useposix=true, d_sigaction=define
     usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
     useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
     use64bitint=undef use64bitall=undef uselongdouble=undef
     usemymalloc=n, bincompat5005=undef
   Compiler:
     cc='cc', ccflags ='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include 
-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
     optimize='-g',
     cppflags='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include 
-I/usr/include/gdbm'
     ccversion='', gccversion='3.3.1 (Mandrake Linux 9.2 3.3.1-2mdk)', 
gccosandvers=''
     intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
     d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
     ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', 
lseeksize=8
     alignbytes=4, prototype=define
   Linker and Libraries:
     ld='cc', ldflags =' -L/usr/local/lib'
     libpth=/usr/local/lib /lib /usr/lib
     libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc
     perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
     libc=/lib/libc-2.3.2.so, so=so, useshrplib=true, libperl=libperl.so
     gnulibc_version='2.3.2'
   Dynamic Linking:
     dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic 
-Wl,-rpath,/home/stas/perl/5.8.1/lib/5.8.1/i686-linux/CORE'
     cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:



@INC for perl v5.8.1:
     /home/stas/perl/5.8.1/lib/5.8.1/i686-linux
     /home/stas/perl/5.8.1/lib/5.8.1
     /home/stas/perl/5.8.1/lib/site_perl/5.8.1/i686-linux
     /home/stas/perl/5.8.1/lib/site_perl/5.8.1
     /home/stas/perl/5.8.1/lib/site_perl
     .


Environment for perl v5.8.1:
     HOME=/home/stas
     LANG=en_AU
     LANGUAGE=en_AU:en_US:en
     LC_ADDRESS=en_AU
     LC_COLLATE=en_AU
     LC_CTYPE=en_AU
     LC_IDENTIFICATION=en_AU
     LC_MEASUREMENT=en_AU
     LC_MESSAGES=en_AU
     LC_MONETARY=en_AU
     LC_NAME=en_AU
     LC_NUMERIC=en_AU
     LC_PAPER=en_AU
     LC_TELEPHONE=en_AU
     LC_TIME=en_AU
     LD_LIBRARY_PATH (unset)
     LOGDIR (unset)
 
PATH=/usr//bin:/bin:/usr/bin:.:/usr/local/bin:/usr/X11R6/bin:/usr/games:/home/stas/bin:/home/stas/bin:/usr/local/bin:/usr/X11R6/bin:/usr/java/j2re1.4.0/bin/
     PERLDOC_PAGER=less -R
     PERL_BADLANG (unset)
     SHELL=/bin/tcsh

__________________________________________________________________
Stas Bekman            JAm_pH ------> Just Another mod_perl Hacker
http://stason.org/     mod_perl Guide ---> http://perl.apache.org
mailto:stas@stason.org http://use.perl.org http://apacheweek.com
http://modperlbook.org http://apache.org   http://ticketmaster.com


```
</details>

**Comment by p5pRT** [comment]
### From @lizmat

At 21&#8203;:30 \+0000 10/15/03\, Stas Bekman \(via RT\) wrote&#8203;:
>% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
>Segmentation fault

I can NOT reproduce this on Linux\, nor Mac OS X\, with either 
threaded\, unthreaded or debug versions of Perl\.

Linux bibi 2\.4\.10\-4GB \#1 Fri Sep 28 17&#8203;:20&#8203;:21 GMT 2001 i686 unknown

Darwin Lyzibook 6\.8 Darwin Kernel Version 6\.8&#8203;: Wed Sep 10 15&#8203;:20&#8203;:55 
PDT 2003; root&#8203;:xnu/xnu\-344\.49\.obj~2/RELEASE\_PPC  Power Macintosh 
powerpc


Maybe this is a Locale setting dependent phenomenon?


Liz

**Comment by p5pRT** [comment]
### From stas@stason.org

Elizabeth Mattijsen \(via RT\) wrote&#8203;:
> At 21&#8203;:30 \+0000 10/15/03\, Stas Bekman \(via RT\) wrote&#8203;:
> 
>>% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
>>Segmentation fault
> 
> 
> I can NOT reproduce this on Linux\, nor Mac OS X\, with either 
> threaded\, unthreaded or debug versions of Perl\.
> 
> Linux bibi 2\.4\.10\-4GB \#1 Fri Sep 28 17&#8203;:20&#8203;:21 GMT 2001 i686 unknown
> 
> Darwin Lyzibook 6\.8 Darwin Kernel Version 6\.8&#8203;: Wed Sep 10 15&#8203;:20&#8203;:55 
> PDT 2003; root&#8203;:xnu/xnu\-344\.49\.obj~2/RELEASE\_PPC  Power Macintosh 
> powerpc
> 
> 
> Maybe this is a Locale setting dependent phenomenon?

You mean some specific Locale\, I've tried changing it but I still reproduce 
the segfault\, with 5\.8\.1\, maint and blead perls\.

Linux rabbit\.stason\.org 2\.4\.21\-0\.18mdkcustom \#6 Mon Jun 16 16&#8203;:26&#8203;:34 EST 2003 
i686 unknown unknown GNU/Linux

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
Stas Bekman            JAm\_pH \-\-\-\-\-\-> Just Another mod\_perl Hacker
http&#8203;://stason\.org/     mod\_perl Guide \-\-\-> http&#8203;://perl\.apache\.org
mailto&#8203;:stas@&#8203;stason\.org http&#8203;://use\.perl\.org http&#8203;://apacheweek\.com
http&#8203;://modperlbook\.org http&#8203;://apache\.org   http&#8203;://ticketmaster\.com

**Comment by p5pRT** [comment]
### From @eserte

Elizabeth Mattijsen \<liz@&#8203;dijkmat\.nl> writes&#8203;:

> At 21&#8203;:30 \+0000 10/15/03\, Stas Bekman \(via RT\) wrote&#8203;:
> >% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
> >Segmentation fault
> 
> I can NOT reproduce this on Linux\, nor Mac OS X\, with either 
> threaded\, unthreaded or debug versions of Perl\.
> 
> Linux bibi 2\.4\.10\-4GB \#1 Fri Sep 28 17&#8203;:20&#8203;:21 GMT 2001 i686 unknown
> 
> Darwin Lyzibook 6\.8 Darwin Kernel Version 6\.8&#8203;: Wed Sep 10 15&#8203;:20&#8203;:55 
> PDT 2003; root&#8203;:xnu/xnu\-344\.49\.obj~2/RELEASE\_PPC  Power Macintosh 
> powerpc
> 
> 
> Maybe this is a Locale setting dependent phenomenon?
> 

It's reproducible if calling with PERLIO=perlio\. This patch seems to
fix it&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- bleedperl/perlio.c	Wed Sep 24 20:47:51 2003
+++ bleedperl2/perlio.c	Sun Oct 19 17:51:10 2003
@@ -2613,8 +2613,10 @@ char *
 PerlIOStdio_mode(const char *mode, char *tmode)
 {
     char *ret = tmode;
-    while (*mode) {
-	*tmode++ = *mode++;
+    if (mode) {
+	while (*mode) {
+	    *tmode++ = *mode++;
+	}
     }
 #if defined(PERLIO_USING_CRLF) || defined(__CYGWIN__)
     *tmode++ = 'b';


-- 
```
</details>

Slaven Rezic \- slaven@&#8203;rezic\.de

&nbsp;    tktimex \- project time manager
&nbsp;    http&#8203;://sourceforge\.net/projects/ptktools/

**Comment by p5pRT** [comment]
### From nick.ing-simmons@elixent.com

Stas Bekman \<perl5\-porters@&#8203;perl\.org> writes&#8203;:
>\# New Ticket Created by  Stas Bekman 
>\# Please include the string&#8203;:  \[perl \#24225\]
>\# in the subject line of all future correspondence about this issue\. 
>\# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt2/Ticket/Display\.html?id=24225 >
>
>
>
>
>This is a bug report for perl from stas@&#8203;stason\.org\,
>generated with the help of perlbug 1\.34 running under perl v5\.8\.1\.
>
>
>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>\[Please enter your report here\]
>
>% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
>Segmentation fault
>
>The same problem appears with blead perl\, but not with 5\.8\.0\.

Confirmed\. Something odd is going on\.\.\.
You are not supposed to get to the else branch on line 2640 if
stream is already stdio\. 

That said blindly de\-refing mode is not good\.

>
>\#0  0x40123024 in PerlIOStdio\_mode \(mode=0x0\,
>     tmode=0xbffff208 "ÃÃ\\001@&#8203;0Ã\\b@&#8203;$Ã²Ã¿Â¿\\001"\) at perlio\.c&#8203;:2616
>2616        while \(\*mode\) \{
>\(gdb\) where
>\#0  0x40123024 in PerlIOStdio\_mode \(mode=0x0\,
>     tmode=0xbffff208 "ÃÃ\\001@&#8203;0Ã\\b@&#8203;$Ã²Ã¿Â¿\\001"\) at perlio\.c&#8203;:2616
>\#1  0x401230e9 in PerlIOStdio\_pushed \(f=0x8054900\, mode=0x0\,
>arg=0x40157618\,
>     tab=0x40155a60\) at perlio\.c&#8203;:2640
>\#2  0x4012086d in PerlIO\_push \(f=0x8054900\, tab=0x40155a60\, mode=0x0\,
>     arg=0x40157618\) at perlio\.c&#8203;:1165
>\#3  0x40120abc in PerlIO\_apply\_layera \(f=0x8054900\, mode=0x0\,
>     layers=0x8054ad8\, n=0\, max=1\) at perlio\.c&#8203;:1248
>\#4  0x40120b3c in PerlIO\_apply\_layers \(f=0x8054900\, mode=0x0\,
>     names=0x805d310 "&#8203;:stdio"\) at perlio\.c&#8203;:1266
>\#5  0x40120bbf in PerlIO\_binmode \(f=0x8054900\, iotype=62\, mode=0\,
>     names=0x805d310 "&#8203;:stdio"\) at perlio\.c&#8203;:1291
>\#6  0x400eae56 in Perl\_pp\_binmode \(\) at pp\_sys\.c&#8203;:755
>\#7  0x4008d1f4 in Perl\_runops\_debug \(\) at dump\.c&#8203;:1434
>\#8  0x400383c4 in S\_run\_body \(oldscope=1\) at perl\.c&#8203;:1829
>\#9  0x40037e81 in perl\_run \(my\_perl=0x804baf8\) at perl\.c&#8203;:1748
>\#10 0x080492b2 in main \(argc=3\, argv=0xbffff504\, env=0xbffff514\)
>     at perlmain\.c&#8203;:86
>\#11 0x401ebc57 in \_\_libc\_start\_main \(\) from /lib/i686/libc\.so\.6
>
>
>\[Please do not change anything below this line\]
>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>\-\-\-
>Flags&#8203;:
>     category=core
>     severity=medium
>\-\-\-
>Site configuration information for perl v5\.8\.1&#8203;:
>
>Configured by stas at Thu Sep 25 19&#8203;:12&#8203;:52 PDT 2003\.
>
>Summary of my perl5 \(revision 5\.0 version 8 subversion 1\) configuration&#8203;:
>   Platform&#8203;:
>     osname=linux\, osvers=2\.4\.21\-0\.18mdkcustom\, archname=i686\-linux
>     uname='linux rabbit\.stason\.org 2\.4\.21\-0\.18mdkcustom \#6 mon jun 16 
>16&#8203;:26&#8203;:34 est 2003 i686 unknown unknown gnulinux '
>     config\_args='\-des \-Dprefix=/home/stas/perl/5\.8\.1 \-Doptimize=\-g 
>\-Duseshrplib \-Dusedevel'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
>     useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>     use64bitint=undef use64bitall=undef uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-DDEBUGGING \-fno\-strict\-aliasing \-I/usr/local/include 
>\-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-I/usr/include/gdbm'\,
>     optimize='\-g'\,
>     cppflags='\-DDEBUGGING \-fno\-strict\-aliasing \-I/usr/local/include 
>\-I/usr/include/gdbm'
>     ccversion=''\, gccversion='3\.3\.1 \(Mandrake Linux 9\.2 3\.3\.1\-2mdk\)'\, 
>gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, 
>lseeksize=8
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-L/usr/local/lib'
>     libpth=/usr/local/lib /lib /usr/lib
>     libs=\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lc
>     perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc
>     libc=/lib/libc\-2\.3\.2\.so\, so=so\, useshrplib=true\, libperl=libperl\.so
>     gnulibc\_version='2\.3\.2'
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic 
>\-Wl\,\-rpath\,/home/stas/perl/5\.8\.1/lib/5\.8\.1/i686\-linux/CORE'
>     cccdlflags='\-fpic'\, lddlflags='\-shared \-L/usr/local/lib'
>
>Locally applied patches&#8203;:
>
>
>\-\-\-
>@&#8203;INC for perl v5\.8\.1&#8203;:
>     /home/stas/perl/5\.8\.1/lib/5\.8\.1/i686\-linux
>     /home/stas/perl/5\.8\.1/lib/5\.8\.1
>     /home/stas/perl/5\.8\.1/lib/site\_perl/5\.8\.1/i686\-linux
>     /home/stas/perl/5\.8\.1/lib/site\_perl/5\.8\.1
>     /home/stas/perl/5\.8\.1/lib/site\_perl
>     \.
>
>\-\-\-
>Environment for perl v5\.8\.1&#8203;:
>     HOME=/home/stas
>     LANG=en\_AU
>     LANGUAGE=en\_AU&#8203;:en\_US&#8203;:en
>     LC\_ADDRESS=en\_AU
>     LC\_COLLATE=en\_AU
>     LC\_CTYPE=en\_AU
>     LC\_IDENTIFICATION=en\_AU
>     LC\_MEASUREMENT=en\_AU
>     LC\_MESSAGES=en\_AU
>     LC\_MONETARY=en\_AU
>     LC\_NAME=en\_AU
>     LC\_NUMERIC=en\_AU
>     LC\_PAPER=en\_AU
>     LC\_TELEPHONE=en\_AU
>     LC\_TIME=en\_AU
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
> 
>PATH=/usr//bin&#8203;:/bin&#8203;:/usr/bin&#8203;:\.&#8203;:/usr/local/bin&#8203;:/usr/X11R6/bin&#8203;:/usr/games&#8203;:/home/stas/bin&#8203;:/home/stas/bin&#8203;:/usr/local/bin&#8203;:/usr/X11R6/bin&#8203;:/usr/java/j2re1\.4\.0/bin/
>     PERLDOC\_PAGER=less \-R
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/tcsh
>
>\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
>Stas Bekman            JAm\_pH \-\-\-\-\-\-> Just Another mod\_perl Hacker
>http&#8203;://stason\.org/     mod\_perl Guide \-\-\-> http&#8203;://perl\.apache\.org
>mailto&#8203;:stas@&#8203;stason\.org http&#8203;://use\.perl\.org http&#8203;://apacheweek\.com
>http&#8203;://modperlbook\.org http&#8203;://apache\.org   http&#8203;://ticketmaster\.com

**Comment by p5pRT** [comment]
### From stas@stason.org

Nick Ing\-Simmons wrote&#8203;:
> Stas Bekman \<perl5\-porters@&#8203;perl\.org> writes&#8203;:
> 
>>\# New Ticket Created by  Stas Bekman 
>>\# Please include the string&#8203;:  \[perl \#24225\]
>>\# in the subject line of all future correspondence about this issue\. 
>>\# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt2/Ticket/Display\.html?id=24225 >
>>
>>
>>
>>
>>This is a bug report for perl from stas@&#8203;stason\.org\,
>>generated with the help of perlbug 1\.34 running under perl v5\.8\.1\.
>>
>>
>>\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>>\[Please enter your report here\]
>>
>>% perl\-5\.8\.1 \-wle "binmode STDOUT\, '&#8203;:stdio'; print 1"
>>Segmentation fault
>>
>>The same problem appears with blead perl\, but not with 5\.8\.0\.
> 
> 
> Confirmed\. Something odd is going on\.\.\.
> You are not supposed to get to the else branch on line 2640 if
> stream is already stdio\. 
> 
> That said blindly de\-refing mode is not good\.

Certainly\, I was just reporting on behalf of someone else\.


\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
Stas Bekman            JAm\_pH \-\-\-\-\-\-> Just Another mod\_perl Hacker
http&#8203;://stason\.org/     mod\_perl Guide \-\-\-> http&#8203;://perl\.apache\.org
mailto&#8203;:stas@&#8203;stason\.org http&#8203;://use\.perl\.org http&#8203;://apacheweek\.com
http&#8203;://modperlbook\.org http&#8203;://apache\.org   http&#8203;://ticketmaster\.com

**Comment by p5pRT** [comment]
### From @rgs

Slaven Rezic wrote&#8203;:
> It's reproducible if calling with PERLIO=perlio\. This patch seems to
> fix it&#8203;:
> 
> \-\-\- bleedperl/perlio\.c    Wed Sep 24 20&#8203;:47&#8203;:51 2003
> \+\+\+ bleedperl2/perlio\.c    Sun Oct 19 17&#8203;:51&#8203;:10 2003

Thanks\, applied as 21505\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'new' to 'resolved'
